<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="Truck" type="system">
        <hardware>
          <plugin>ros2_control_demo_example_11/CarlikeBotSystemHardware</plugin>
          <param name="example_param_hw_start_duration_sec">0</param>
          <param name="example_param_hw_stop_duration_sec">3.0</param>
          <!-- Define parameters for joint limits -->
          <param name="steering_position_min">-0.5724</param>
          <param name="steering_position_max">0.5724</param>
          <param name="traction_velocity_min">-11.1111</param>
          <param name="traction_velocity_max">11.1111</param>
          <param name="device">/dev/ttyAMA10</param>
          <param name="baud_rate">460800</param>
          <param name="timeout">200</param>
        </hardware>
        <joint name="virtual_front_wheel_joint">
          <command_interface name="position"/>
          <state_interface name="position"/>
        </joint>
        <joint name="virtual_rear_wheel_joint">
          <command_interface name="velocity"/>
          <state_interface name="velocity"/>
          <state_interface name="position"/>
        </joint>
      </ros2_control>

</robot>